### PR TITLE
build with ES6

### PR DESCRIPTION
**Pull request by kwhumphreys**

### CODE REVIEW

**Comment by kwhumphreys** [comment]
@dgingrich any pointers on how to publish this fork to pip-s3?

File: requirements.txt
Comment (dgingrich) [inline requirements.txt:1]: Wow, that's a huge version jump.  Looks like the old elasticmock is just really out of date?

File: elasticmock/fake_elasticsearch.py
Comment (dgingrich) [inline elasticmock/fake_elasticsearch.py:237]: Why remove the unicode check?  If we want to merge this upstream we should minimize the changes.

File: setup.py
Comment (dgingrich) [inline setup.py:20]: Same here.  Requiring it to be >=6.0.0 is sort of limiting.  Do we really require that, or can we get away with a lower version?

**Review by dgingrich — APPROVED** [review]
Changes look good.  As for getting it into PyPI: easiest is to get the changes adopted upstream. 

If we have to push to pypi-s3 in a semi-permanent way, the best is to add a Makefile that does the normal `deploy`, `release`, `deploy-prod` steps to this repo.  

If we just need to have a version in pypi-s3 for awhile (until the upstream merge is done) and we don't expect to make many changes, we can manually create the package and just `aws s3 cp` it into the right pypi bucket.  We should also document those instructions in the README in case we have to repeat them.

In either case we probably want to upload under a slightly different package name, like `textio-elasticmock`, to make sure we pull down our version and not the prod one.  If that doesn't work for some reason (like it's a transitive dependency of another package) we probably want to bump the major version number to make it less likely we'd accidentally collide.

Let me know if you want help with the pypi-s3 stuff.

File: requirements.txt
Comment (kwhumphreys) [reply requirements.txt:1]: yes, so probably more changes will be needed, and an upstream PR probably won't be accepted until coverage is complete, but for now I'm just hacking the parts we need to use

File: elasticmock/fake_elasticsearch.py
Comment (kwhumphreys) [reply elasticmock/fake_elasticsearch.py:237]: `unicode` is a Python 2 type. not needed (and doesn't compile) for Python 3.

File: setup.py
Comment (kwhumphreys) [reply setup.py:20]: probably, but we haven't used less that v6 for FFwd, and v6 did introduce some ES API changes

Comment (kwhumphreys) [reply setup.py:20]: it would also mean testing a bunch of ES versions which we don't use

**Comment by dgingrich** [comment]
If we think we can get merged upstream we'll have to do things like use `six` to make the string handling (`unicode` check) work across Python 2 and 3, make the `elasticsearch` version flexible, etc.  That's a lot of extra work, but I doubt we'll get a patch adopted without it (unless the maintainer wants to supports lots of different major versions for Py2/3 and elasticsearch versions).  If this is what we want we should open an issue in the upstream repo.

If we're just hacking on this instead of writing our own mocking, then the package rename is better.  As for how to get it into pypi-s3, we have the same choices: update Makefile to upload to pypi-s3, i.e. make this build like our other packages; or do a one-off `aws s3 cp`.

**Pull request merged**
